File: app/base_code/src/supervisors/events.c
#include <stdlib.h>

#include "../../lib/supervisor/supervisor.h"

// create events
Event btn = {UNCONTROLLABLE, 0, SUP_DEBUG_STR("btn"), NULL};
Event liga = {CONTROLLABLE, 1, SUP_DEBUG_STR("liga"), NULL};
Event desliga = {CONTROLLABLE, 2, SUP_DEBUG_STR("desliga"), NULL};

File: app_python/base_code/Supervisor/events.py
"""
Eventos lidos no arquivo do supremica
"""
from . import Event, EventKind

# Create events
Events = {
    'btn': Event(EventKind.UNCONTROLLABLE, 0, 'btn'),
    'liga': Event(EventKind.CONTROLLABLE, 1, 'liga'),
    'desliga': Event(EventKind.CONTROLLABLE, 2, 'desliga'),
}

def get_event_by_id(event_id: int) -> Event:
    """
    Get event by id
    """
    if not isinstance(event_id, int):
        raise TypeError("event_id must be an integer")

    for event in Events.values():
        if event.get_id() == event_id:
            return event
    return None